pypeit/io.py

```diff
@@ -250,7 +252,10 @@ def initialize_header(hdr=None):
     # the versions of all packages included in the requirements.txt
     # file?
     if hdr is None:
-        hdr = fits.Header()
+        if primary:
```

Comment:
Why is this necessary?  I played with this a bit and I may understand why you did this, but did you see some behavior where astropy wasn't treating primary vs non-primary headers correctly when the `HDUList` was instantiated?

Reply:
I have to admit my memory is fuzzy.  I think it was
in a corner case of re-running pypeit (e.g. when 
we are adding to an existing `spec1d` file.).

Reply:
considered resolved